spinalcordtoolbox/utils.py

```diff
@@ -115,6 +116,23 @@ def _split_lines(self, text, width):
         return argparse.HelpFormatter._split_lines(self, text, width)
 
 
+def add_suffix(fname, suffix):
+    """
+    Add suffix between end of file name and extension.
+
+    :param fname: absolute or relative file name. Example: t2.nii
+    :param suffix: suffix. Example: _mean
+    :return: file name with suffix. Example: t2_mean.nii
+
+    Examples:
+
+    - add_suffix(t2.nii, _mean) -> t2_mean.nii
+    - add_suffix(t2.nii.gz, a) -> t2a.nii.gz
+    """
+    parent, stem, ext = extract_fname(fname)
+    return os.path.join(parent, stem + suffix + ext)
```

Comment:
You don't need to separate the parent:

```
>>> base, ext = os.path.splitext("path/to/dataset/t2.nii.gz")
>>> base + '_mean' + ext
'path/to/dataset/t2.nii_mean.gz'
```

If that was `sct.utils.splitext()` instead that was `.nii.gz`-aware that would have produced the right result.

Reply:
It handles `.`s in the parent dirs properly:

```
>>> base, ext = os.path.splitext("path/to.tar.gz/dataset/t2.tar.gz")
>>> base + "_mean" + ext
'path/to.tar.gz/dataset/t2.tar_mean.gz'
>>> 
```